app/controllers/bot_instances_controller.rb

```diff
@@ -2,18 +2,21 @@ class BotInstancesController < ApplicationController
   before_action :authenticate_user!, :only => [:new, :create, :edit, :update, :destroy]
   before_action :set_bot_instance, only: [:show, :edit, :update, :destroy]
   before_action :check_instance_permissions, only: [:edit, :update, :destroy]
-  before_action :check_bot_permissions, only: [:new, :create]
```

Comment:
I needed `check_bot_permissions` to have access to `@bot` so it could check for permissions.

Why does `and return` work?  Doesn't that just return from `check_bot_permissions`?